include/spdlog/sinks/minute_file_sink.h

```diff
@@ -0,0 +1,217 @@
+// Copyright(c) 2015-present, Gabi Melman & spdlog contributors.
+// Distributed under the MIT License (http://opensource.org/licenses/MIT)
+//File: minute_file_sink to support every 1 min file rotation.
+//Author Manoj Mallawaarachchie manoj_ws@yahoo.com / manoj@vizuamatix.com
+//example driver code:
+//auto logger = spdlog::minute_logger_mt("minutes_basic_logger", "logs/min-log.txt",false,60);
+
+#pragma once
+
+#include <spdlog/common.h>
+#include <spdlog/details/file_helper.h>
+#include <spdlog/details/null_mutex.h>
+#include <spdlog/fmt/fmt.h>
+#include <spdlog/sinks/base_sink.h>
+#include <spdlog/details/os.h>
+#include <spdlog/details/circular_q.h>
+#include <spdlog/details/synchronous_factory.h>
+
+#include <chrono>
+#include <cstdio>
+#include <ctime>
+#include <mutex>
+#include <string>
+
+namespace spdlog {
+namespace sinks {
+
+/*
+ * Generator of Minute log file names in format basename.YYYY-MM-DD-HH-MM.ext
+ */
+struct minute_filename_calculator
+{
+    // Create filename for the form basename.YYYY-MM-DD-H-M
+    static filename_t calc_filename(const filename_t &filename, const tm &now_tm)
+    {
+        filename_t basename, ext;
+        std::tie(basename, ext) = details::file_helper::split_by_extension(filename);
+        return fmt_lib::format(SPDLOG_FMT_STRING(SPDLOG_FILENAME_T("{}_{:04d}-{:02d}-{:02d}-{:02d}_{:02d}{}")), basename, now_tm.tm_year + 1900, now_tm.tm_mon + 1,
+            now_tm.tm_mday, now_tm.tm_hour,now_tm.tm_min,ext);
+    }
+};
+
+/*
+ * Rotating file sink based on time(minutes).
+ * If truncate != false , the created file will be truncated.
+ * If max_files > 0, retain only the last max_files and delete previous.
+ */
+template<typename Mutex, typename FileNameCalc = minute_filename_calculator>
+class minute_file_sink final : public base_sink<Mutex>
+{
+public:
+   // create every minute file sink which rotates on given time
+    minute_file_sink(
+        filename_t base_filename, bool truncate = false, uint16_t max_files = 0,
+        int rotation_minute = 0, const file_event_handlers &event_handlers = {})
+        : base_filename_(std::move(base_filename))
+        , file_helper_{event_handlers}
+        , truncate_(truncate)
+        , max_files_(max_files)
+        ,rotation_m_(rotation_minute)
+        , filenames_q_()
+    {
+        
+        if (rotation_minute < 0 || rotation_minute > 59)
+        {
+            throw_spdlog_ex("daily_file_sink: Invalid rotation time in ctor");
```

Comment:
should be "minute_file_sink"

Reply:
@gabime 

Thanks for the feedback and got your point, remove the validation considering the concern.

Reply:
@mmanoj Note: In the current rotation implementation, `rotation_m_` should not be allowed to be `0`.

Reply:
@tt4g 
Implement the check for 0 min.